### ISSUE TITLE
How to generate normals?

### ISSUE THREAD

**Issue by weiforce** [hzhao1997/HF-Avatar#8]
Nice work. But I ran into difficulties.
I have prepared other data files, but I don't know how to generate normal files with pifuhd. 
The demo of pifuhd only generates obj.

**Comment by hzhao1997** [comment]
please see this: https://github.com/facebookresearch/pifuhd/blob/ee51fc23629c4afc99dbfd235c5b8bbd77a86aa2/apps/recon.py#L43, we save `net.netG.nmlF` to .npz directly by
`front_normal = np.transpose(net.netG.nmlF[0].detach().cpu().numpy(), (1, 2, 0))`
`np.savez_compressed('front', front_normal)`

**Comment by weiforce** [comment]
thanks, it worked.

**Comment by Gary-Qinghua** [comment]
如果采用这种方式生成的normal，它的尺寸不太对
如图，第一张是我用作者提供的frame_mat生成的normal
![b](https://user-images.githubusercontent.com/73276075/206612867-11cf510b-2ea7-4012-815c-b544cd4fc1d0.jpg)
第二张是作者提供的normal里面的
![a](https://user-images.githubusercontent.com/73276075/206612932-42d4243b-ee00-40ac-9348-ea53d6f82f15.jpg)
很明显尺寸不对，第一张如果用mask_mat进行背景剔除的话，产生的图片是有问题的

**Comment by Gary-Qinghua** [comment]
将pifuhd数据预处理部分的
im = crop_image(im, rect)  注释掉就跟作者提供的一模一样了

**Comment by Xv-M-S** [comment]
where is the code of "im = crop_image(im, rect)"

**Comment by Marianoeli** [comment]
I can't find "im = crop_image(im, rect) ",too

**Comment by Xv-M-S** [comment]
I finally find it in there https://github.com/facebookresearch/pifuhd/blob/a9fa3b35754d261bd1e12dfb1b09b91ce4547d92/lib/data/EvalWPoseDataset.py#L244

**Comment by Marianoeli** [comment]
I never thought that, truly appreciate it,thanks!!

> I finally find it in there https://github.com/facebookresearch/pifuhd/blob/a9fa3b35754d261bd1e12dfb1b09b91ce4547d92/lib/data/EvalWPoseDataset.py#L244

**Comment by Marianoeli** [comment]
excuse me, another question:what exact instruction should I run for pifuhd given that I get 800 frame pictures whereas it takes least 4 minutes to process per picture? Should I comment some code lines or what?

**Comment by Xv-M-S** [comment]
I‘m not quite sure about it , you should add some code to generate .npz file as the author noted in there https://github.com/hzhao1997/HF-Avatar/issues/8#issuecomment-1326072140 . As for how to run pifuhd , you can see here  https://github.com/facebookresearch/pifuhd#testing for details . And it surly takes some time to process per picture.

**Comment by Marianoeli** [comment]
I'll try to figure it out

**Comment by Marianoeli** [comment]
> please see this: https://github.com/facebookresearch/pifuhd/blob/ee51fc23629c4afc99dbfd235c5b8bbd77a86aa2/apps/recon.py#L43, we save `net.netG.nmlF` to .npz directly by `front_normal = np.transpose(net.netG.nmlF[0].detach().cpu().numpy(), (1, 2, 0))` `np.savez_compressed('front', front_normal)`

Is there any concrete code can be displayed?
![image](https://user-images.githubusercontent.com/112232967/226150688-6fb3ebaf-1c07-4efb-8987-7e6202af9132.png)
![image](https://user-images.githubusercontent.com/112232967/226150701-999439a1-22f3-4e91-a49e-f2fdec48a9cf.png)
Additionally,what should I do if I just want normal files to be generated to the specified folder without generating .obj files?

**Comment by Xv-M-S** [comment]
Sorry,I'm also not quite famliy with it.

I modified like this:

try:
        if net.netG.netF is not None:
            image_tensor_global = torch.cat([image_tensor_global, net.netG.nmlF], 0)
            #save .npz file
            front_normal = np.transpose(net.netG.nmlF[0].detach().cpu().numpy(), (1, 2, 0))
            save_normal_path = save_path[:-4]
            np.savez_compressed(save_normal_path, front_normal)
            #save .npz file 
        if net.netG.netB is not None:
            image_tensor_global = torch.cat([image_tensor_global, net.netG.nmlB], 0)

I guess you can annotate code [https://github.com/facebookresearch/pifuhd/blob/ee51fc23629c4afc99dbfd235c5b8bbd77a86aa2/apps/recon.py#L79](url)   to generate the specified folder without generating .obj files
